docs/mp/config/02_MP_config_sources.adoc

```diff
@@ -0,0 +1,155 @@
+	}
+
+}
+----
+
+<1> Returns the name of the Config Source to use for logging or analysis of configured values.
+<2> Returns the properties in this Config Source as a map.
+<3> Returns the value stored in the propertyName of this Config Source.
+<4> Returns the ordinal based on the set order for this Config Source.
+
+
+== Creating MicroProfile Config Sources for Manual Setup of Config
+
+You can use the following methods to create MicroProfile Config Sources to manually set up the Config from `org.eclipse.microprofile.config.spi.ConfigProviderResolver#getBuilder()` on `io.helidon.config.mp.MpConfigSources` class:
+
+|===
+|Method |Description
+
+|`systemProperties()}`   |System properties config source.
+
+|`environmentVariables()}`   |Environment variables config source.
+
+|`create(java.nio.file.Path)}`   |Loads a properties file from file system. +
+To load the properties file from file system with custom name, use `create(String, java.nio.file.Path)}`.
+
+|`create(java.util.Map)}`   |Creates an in-memory source from map. +
+To create an in-memory source from map with custom name, use `create(String, java.util.Map)}`.
+
+|`create(java.util.Properties)}`   |Creates an in-memory source from properties. +
+To create an in-memory source from properties with custom name, use `create(String, java.util.Properties)}`.
+
+|===
+
+=== Example
+
+[source,java]
+.Create MicroProfile Config Source based on Environment Variables and Custom Map
+----
+org.eclipse.microprofile.config.Config mpConfig = ConfigProviderResolver.instance()
+        .getBuilder() <1>
+        .withSources(MpConfigSources.environmentVariables()) <2>
+        .withSources(MpConfigSources.create(Map.of("key","value"))) <3>
+        .build(); <4>
+----
+<1> Creates MicroProfile Config Source builder.
+<2> Adds environment variables.
+<3> Adds custom map.
+<4> Builds the MicroProfile Config Source.
+
+
+== Creating MicroProfile Config Source from Helidon SE Config Source
+
+You can create MicroProfile Config Source from Helidon SE Config Source to use Helidon SE features in Helidon MP. The Config Source is immutable regardless of configured polling strategy or change watchers.
+
+
+// Please provide example
+
+
+== Creating MicroProfile Config Source from Helidon SE Config Instance
+
+You can create MicroProfile Config Source from Helidon SE Config to use advanced Helidon SE features in Helidon MP. The Config Source is mutable if the config uses either polling strategy and change watchers or polling strategy or change watchers.
+The latest config version is queried each time  `org.eclipse.microprofile.config.spi.ConfigSource#getValue(String)` is called.
+
+//Please provide example
+
+== Create Yaml MicroProfile Config Source Programatically
+
+You can create Yaml Microprofile Config Source from a path or a URL. When you create a MicroProfile instance from the builder, the `YamlMPConfigSource` allows you to edit the list of config sources and creates a custom YAML.

```

Comment:
It does not "allow you to edit"
It works exactly the same as the `MpConfigSources` above - you can use `YamlMpConfigSource` (btw there is a typo in your text in class name) to create a new config source to register it with the builder.